packages/augur-sdk/src/state/db/SyncStatus.ts

```diff
@@ -25,12 +28,26 @@ export class SyncStatus extends AbstractDB {
     });
   }
 
-  public async setHighestSyncBlock(dbName: string, blockNumber: number, syncing: boolean): Promise<PouchDB.Core.Response> {
-    const document: SyncDocument = { blockNumber, syncing };
-    return this.upsertDocument(dbName, document);
+  public async setHighestSyncBlock(dbName: string, blockNumber: number, syncing: boolean, rollback: boolean = false): Promise<PouchDB.UpsertResponse> {
+
+    const highestKnownBlock = await this.getHighestSyncBlock(dbName);
+
+    // NOTE: dbName, in this case, is actually the id of the record in the SyncStatus db.
+    return this.db.upsert(dbName, (document: SyncDocument) => {
+      // make sure the truly highest block is always being used
+      if (!rollback) {
```

Comment:
is this just a paranoia thing or would this happen normally?

Reply:
Paranoia and a suggestion by @pgebheim. In a browser this won't happen. In a server maybe.